bundles/org.openhab.binding.mqtt.espmilighthub/README.md

```diff
@@ -0,0 +1,156 @@
+Next make sure you have moved a control on either a physical remote, or used a virtual control inside the esp8266 control panel web page to create a MQTT message.
+This binding should then detect the new device the moment the control is moved and a new entry should appear in your INBOX.
+
+To remove a saved state from your MQTT broker that causes an entry in your INBOX you can use this command or use the ignore feature of openHAB.
+
+```
+mosquitto_pub -u username -P password -p 1883 -t 'milight/states/0x0/rgb_cct/1' -n -r
+```
+
+## Thing Configuration
+
+| Parameter | Description | Required | Default |
+|-|-|-|-|
+| `defaultCommand` | When the send command button is pressed, if no selection is made from the command channel then use this. | Y | night_mode |
+| `whiteHue` | When both the `whiteHue` and `whiteSat` values are seen by the binding it will trigger the white LEDS. Set to -1 to disable, 0 for Alexa, or 36 for Google Home. | Y | 35 |
+| `whiteSat` | When both the whiteHue and whiteSat values are seen by the binding it will trigger the white LEDS. Set to -1 to disable, 100 for Alexa or 32 for Google Home. | Y | 32 |
+| `favouriteWhite` | When one of the shortcuts triggers white mode, use this for the colour white instead of the default colour. | Y |200 |
+| `dimmedCT` | Traditional globes grow warmer the more they are dimmed. Set this to 370, or leave blank to disable. | N | blank |
+| `oneTriggersNightMode` | `Night Mode` is a much lower level of light and this feature allows it to be auto selected when your fader/slider moves to 1%. NOTE: Night mode locks out some controls of a physical remote so this is disabled by default. | Y | false |
+| `powerFailsToMinimum` | If lights loose power from the power switch OR a power outage, they will default to using the lowest brightness if the light was turned off before the power failure occurred. | Y | true |
+| `whiteThreshold` | RGBW globes do not respond to saturation changes, so this feature allows you to specify a number that if the saturation drops below, it will trigger the white mode. -1 will disable this feature. | Y | 12 |
+
+## Channels
+
+| Channel | Type | Description |
+|-|-|-|
+| level | Dimmer | Level changes the brightness of the globe. |
+| colourTemperature | Dimmer | Change from cool to warm white with this control. |
+| colour | Color | Allows you to change the colour, brightness and saturation of the globe. |
+| discoMode | String | Switch to a Disco mode directly from a drop down list. |
+| bulbMode | String (read only) | Displays the mode the bulb is currently in so that rules can determine if the globe is white, a color, disco modes or night mode are selected. |
+| command | String | Choose the command from this list to send to the globe, then press the `sendCommand` button to send it as many times as you wish. |
```

Comment:
It does do that as well. The idea was select the command and it gets sent instantly when selected, then if you want to repeat the command the button sends it as many times as you wish. The result is less controls on a sitemap that allow access to all the commands, which most people will never use the commands.

Reply:
Understood. The problem I see here is that you are implementing an UI feature in the data model, which isn't really a good approach. Depending on the UI, the commands are rendered as individual buttons. I think you refer to a drop down field?

Reply:
I use a sitemap with a selection item for testing, but I actually never use these myself. Having looked at the commands, the only ones which can not be done any other way is mode_speed_up and mode_speed_down. 

So I guess the choices are:
1. Throw away the two channels and remove the code.
2. Leave as is.

If the code is thrown away it can then be added by anyone in the future to move to using trigger? channels which is probably the way forward having not used them before.

Reply:
Did you try `Switch`?
`Switch item=Dusk_LED label="Sonne"` is rendered like this in Basic UI:
![grafik](https://user-images.githubusercontent.com/1894841/102191572-e2ceaf00-3eb9-11eb-9909-dde0f58cf62e.png)

Would that be practical for your use case?

Reply:
Yes that would be practical as it is a 1 way protocol, just like a dumb IR remote. The switch should be a momentary action and not a latching ON and OFF state. What is the best way to implement that so it appears in the new UI by default? NOTE I changed the channels to Switch types, however I am thinking you may be suggesting another way?

Reply:
To achieve the button view as in above screenshot, you need to make the Channel of type `String` and define state options in the XML file. Each state option will be rendered to a button. In the `.sitemap` file, the type must be `Switch`: `Switch item=Dusk_LED label="Sonne"`

Reply:
I made the change and it works as a drop down list in the new V3 UI, I tried to get the switches as per your post by using the UI to change the item to a Switch and all it gave was a ON and OFF ability. Anyway it works, I'm happy with it as I don't use the feature. Thanks for your help with this area and also doing the review.

Reply:
Ok, I'm not aware that there is any difference in Basic UI concerning the rendering of buttons between 2.5 and 3.x...